pkg/repository/v1/task.go

```diff
@@ -1003,8 +1010,9 @@ func (r *TaskRepositoryImpl) ProcessTaskTimeouts(ctx context.Context, tenantId s
 				InsertedAt: task.InsertedAt,
 				RetryCount: task.RetryCount,
 			},
-			IsAppError:   true,
-			ErrorMessage: fmt.Sprintf("Task exceeded timeout of %s", task.StepTimeout.String),
+			IsAppError:     true,
+			ErrorMessage:   fmt.Sprintf("Task exceeded timeout of %s", task.StepTimeout.String),
+			IsNonRetryable: false,
```

Comment:
Minor nit: Go default value for a boolean is `false`, so this is unnecessary though ok to be explicit